fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Transaction.java

```diff
@@ -96,6 +99,38 @@ public String computeId() {
     return Hash.hash(collectTransaction.toArray(new String[0]));
   }
 
+  /**
+   * Function that given a key pair change the sig of an input considering all the outputs
+   *
+   * @param keyPair of one input sender
+   * @return sig other all the outputs and inputs with the public key
+   */
+  public static String computeSigOutputsPairTxOutHashAndIndex(
+      KeyPair keyPair, List<Output> outputs, Map<String, Integer> inputs_pairs) {
+    // input #1: tx_out_hash Value //input #1: tx_out_index Value
+    // input #2: tx_out_hash Value //input #2: tx_out_index Value ...
+    // TxOut #1: LaoCoin Value​​ //TxOut #1: script.type Value //TxOut #1: script.pubkey_hash Value
+    // TxOut #2: LaoCoin Value​​ //TxOut #2: script.type Value //TxOut #2: script.pubkey_hash
+    // Value...
+    List<String> sig = new ArrayList<>();
+    Iterator<Map.Entry<String, Integer>> ite_input = inputs_pairs.entrySet().iterator();
+    Iterator<Output> ite_output = outputs.iterator();
+
+    while (ite_input.hasNext()) {
+      Map.Entry<String, Integer> current = ite_input.next();
+      sig.add(current.getKey());
+      sig.add(String.valueOf(current.getValue()));
+    }
+
+    while (ite_output.hasNext()) {
+      Output current = ite_output.next();
+      sig.add(String.valueOf(current.getValue()));
+      sig.add(current.getScript().getType());
+      sig.add(current.getScript().getPubkeyHash());
+    }
+    return Hash.hash(sig.toArray(new String[0]));
```

Comment:
>we sign over this entire concatenated string. (final report)

This line should just concatenate everything. Don't need any hashing. In fact you can use a `StringBuilder` if you want.

Reply:
Currently it is not used, but it is used in the https://github.com/dedis/popstellar/pull/1097 following pr.
I can remove it for the moment if you think it is better? and I will see in the next pr.
What do you think ?